pytraj/all_actions.py

```diff
@@ -157,6 +158,33 @@ def _assert_mutable(trajiter):
             "This analysis does not support immutable object. Use `pytraj.Trajectory`"
         )
 
+def pair_distance(p1, p2):
+    x1, y1, z1 = p1
+    x2, y2, z2 = p2
+    return np.sqrt((x1 - x2)**2 + (y1 - y2)**2 + (z1 - z2)**2)
+
+@register_pmap
+def closest_atom(top=None, frame=None, point=(0, 0, 0), mask=""):
+    if (top is None) or (len(top.atom_indices(mask)) == 0):
+        raise ValueError("Please pass in a topology file with atoms that match the mask in it")
+    if (frame is None) or (len(frame.coordinates) == 0):
+        raise ValueError("Please pass in a valid frame with coordinates to the function")
+    if (type(point) is not tuple):
+        raise ValueError("Point argument with xyz coordinates must be a tuple")
+    if len(point) != 3:
+        raise ValueError("Please pass in a tuple of length 3, for xyz coordinates")
+
+    closest_dist = None
+    closest_idx = None
+    atoms = top.atom_indices(mask)
+    for atm in atoms:
+        coord = frame.atom(atm)
+        if ((closest_dist is None) or (pair_distance(coord, point) < closest_dist)):
+            closest_dist = pair_distance(coord, point)
+            closest_idx = atm
```

Comment:
do we care about speed? since this is a python loop calling `pair_distance` many times.
If not (if you're happy with current one), I am good too.
If no, we might want to move the code to Cython to speed up (but it's up to you).

Reply:
I would be open to moving it to Cython, but I don't know how to do that. Maybe we can do it after I've added some of the other water GIST analysis functions that I want to do.

Reply:
Agree. Make it work first and worry about speed later.
(working with cython requires rebuild pytraj and you are having issue with libcpptraj).